### DISCUSSION TITLE
option to change file naming scheme to allow sortable ordering of files

### DISCUSSION THREAD

**Discussion by tedder** [gopro/labs#347] [Feature Requests]
[Gopro chapter numbers are in the filename BEFORE the video number](https://community.gopro.com/s/article/GoPro-Camera-File-Naming-Convention?language=en_US). This means naive sorting does not work properly:

```
GH010333.mp4
GH010444.mp4
GH020333.mp4
GH020444.mp4
GH030333.mp4
GH030444.mp4
GH040333.mp4
GH040444.mp4
```

Compare this to the order the files would have been created:
```
GH010333.mp4
GH020333.mp4
GH030333.mp4
GH040333.mp4
GH010444.mp4
GH020444.mp4
GH030444.mp4
GH040444.mp4
```

[With Labs, base filenames (prefixes)](https://gopro.github.io/labs/control/basename/) can be changed, but this doesn't affect the chapter numbering. It'd be nice to have a labs feature to either fully control the filename, or to have an option to flip the chapter and file numbering.

**Comment by dnewman-gpsw** [comment]
You can increase the chapter size up to 64 GB, so you shouldn't see chaptering in 99% of productions (HERO11 only.)  see https://www.youtube.com/shorts/iQXakUxuYXs

Another solution, closer to your request, is adding date and time to the file name as your prefix.  This enables correct sorting of files, even with the current naming scheme.  

Changing the file and chapter number position is not currently possible, but it is a good idea.

**Reply by tedder** [reply]
yeah, I've been increasing the length, which helps, and I run it through a little postprocessor ([similar to this](https://github.com/AMDmi3/gopro-rename)) to rename them for my sanity.

I didn't think about _time_ being helpful for sorting. Sometimes I see a timestamp that must be pre-gps-acquire, so the time is way off, but those are relatively rare.
